lightning/src/routing/mod.rs

```diff
@@ -26,7 +26,14 @@ use sync::{Mutex, MutexGuard};
 pub trait Score {
 	/// Returns the fee in msats willing to be paid to avoid routing through the given channel
 	/// in the direction from `source` to `target`.
-	fn channel_penalty_msat(&self, short_channel_id: u64, source: &NodeId, target: &NodeId) -> u64;
+	///
+	/// The maximum amount which may be in the HTLC at the given channel is given by
+	/// `send_amt_sat`.
+	///
+	/// The channel's capacity in satoshis (given other HTLCs to be sent) is given by
+	/// `channel_capacity_sats`. It may be guessed from various sources or assumed from no data.
+	/// Your code should be overflow-safe through a `channel_capacity_sats` of 21 million BTC.
+	fn channel_penalty_msat(&self, short_channel_id: u64, send_amt_msat: u64, channel_capacity_sat: u64, source: &NodeId, target: &NodeId) -> u64;
```

Comment:
Is this liquidity or capacity? Not sure if we've been consistent but I thought a channel has a total capacity with inbound and outbound liquidity being part of that. Or does each of those correspond to part of inbound and outbound capacity (i.e. accounting for inflight payments).

Reply:
Capacity, we cannot know liquidity of a third-party channel.